### PR TITLE
[localization] Add default temperature unit to the locale

### PR DESCRIPTION
**Pull request by behenate**
# Why

We want to add a temperature unit field which will return the temperature unit used in the region
ENG-6859

# How

On iOS the unit is based on the current settings in the region tab of the settings app (note: settings values are ignored in iOS simulator so any testing in the future should be done on a physical device)
On Android and web a list of regions which use fahrenheit is used and compared with the current device region.

# Test Plan

Tested in Bare Expo on Android 13, iOS 16 and web

### CODE REVIEW

File: packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationUtils.kt [hunk 1]
```diff
@@ -32,3 +35,17 @@ fun getCurrencyCode(locale: Locale): String? {
     currency?.currencyCode
   }.getOrNull()
 }
+
+fun getRegionCode(locale: Locale): String? {
+  val miuiRegion = getSystemProperty("ro.miui.region")
+  return if (!TextUtils.isEmpty(miuiRegion)) {
```
Comment (lukmccall) [inline packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationUtils.kt:41]:
```suggestion
  return if (!miuiRegion.isEmpty()) {
```
I think you can use `isEmpty` function directly.

**Review by lukmccall — APPROVED** [review]

**Review by aleqsio — APPROVED** [review]
Think this looks good, thanks!

**Review by amandeepmittal — APPROVED** [review]
Thank you for updating the docs for this new addition!

File: packages/expo-localization/ios/LocalizationModule.swift [hunk 2]
```diff
@@ -154,6 +155,20 @@ public class LocalizationModule: Module {
...
+    formatter.locale = Locale.current
+
+    let temperature = Measurement(value: 0, unit: UnitTemperature.celsius)
+    let formatted = formatter.string(from: temperature)
+
+    guard let unitCharacter = formatted.last else {
+      return nil
+    }
+
+    return unitCharacter == "F" ? "fahrenheit" : "celsius"
```
Comment (tsapeta) [inline packages/expo-localization/ios/LocalizationModule.swift:169]: This should also work and is a bit more concise. I think we can assume that if, for some reason, the unit is not present in the formatted string, it should be `celsius`.

```suggestion
    return formatter.last == "F" ? "fahrenheit" : "celsius"
```

File: packages/expo-localization/src/Localization.types.ts [hunk 3]
```diff
@@ -113,6 +113,12 @@ export type Locale = {
    * Ask for user preferences if possible.
    */
   measurementSystem: `metric` | `us` | `uk` | null;
+
+  /**
+   * The temperature unit used in the locale.
+   * Will return null if the region code is unknown.
```
Comment (tsapeta) [inline packages/expo-localization/src/Localization.types.ts:119]:
```suggestion
   * Returns `null` if the region code is unknown.
```

**Review by tsapeta — APPROVED** [review]

**Pull request merged**